### PR TITLE
Programatic graphql config

### PR DESCRIPTION
**Pull request by zth**
This is a draft of adding a way to supply a pre-existing `graphql-config` to the LS. The idea is that external consumers, like a vscode extension, should be able to provide defaults and extensions of the existing config (or generate it programatically all together) in a way that can be supplied directly to the LS.

Feedback much welcome!

### CODE REVIEW

File: packages/graphql-language-service-server/src/GraphQLCache.ts [hunk 1]
```diff
@@ -54,8 +54,11 @@ const {
   DIRECTIVE_DEFINITION,
 } = Kind;
 
-export async function getGraphQLCache(configDir: Uri): Promise<GraphQLCache> {
-  const graphQLConfig = await getGraphQLConfig(configDir);
+export async function getGraphQLCache(
+  configDir: Uri,
+  config?: GraphQLConfig,
+): Promise<GraphQLCache> {
+  const graphQLConfig = config || (await getGraphQLConfig(configDir));
```
Comment (acao) [inline packages/graphql-language-service-server/src/GraphQLCache.ts]: what i was thinking is that if you can pass a config instance here, you might be able to create the graphql-config programatically from an object literal, especially with the new 3.0.0 graphql-config once thats stable

Comment (zth) [reply packages/graphql-language-service-server/src/GraphQLCache.ts]: Not sure I follow what you mean. Could you expand a bit?

Comment (acao) [reply packages/graphql-language-service-server/src/GraphQLCache.ts]: my bad, i'm seeing here where 2.0.0 better suits this need, i was mistaken and thought that 1.2.x supported this too, but i see that this is the only option for now

**Comment by acao** [comment]
@zth we merged the update to 3.0.0 rc 16 so have at it! it will be stable by the time we are. there are a few fantastic new options!

**Comment by acao** [comment]
@zth i think this looks ready, and should work seamlessly with the new graphql-config

**Comment by zth** [comment]
@acao do you want me to resolve the last merge conflict, and then it's good to go for merging?

**Comment by acao** [comment]
@zth yeah if you could, that would be awesome. would love to pull this into the monaco service effort

**Comment by acao** [comment]
@zth ah, we merged another upstream PR, so a bit more to rebase.

**Comment by zth** [comment]
https://github.com/graphql/graphiql/pull/1432 supersedes this.

**Pull request closed without merging**